### DISCUSSION TITLE
Tailwind cli watch not serving localhost webpage on MAC OS X

### DISCUSSION THREAD

**Discussion by DuncanGardiner** [tailwindlabs/tailwindcss#12237] [Help] [answered]
Hi I am building  Django app in VSCode on my development PC an IMAC I5 32GB running Sonoma 14.0 OS X

I am building the site application in Django with a MYSQL backend all contained within a Python Venv virtual machine.

without Tialwindcss installed the site builds and runs perfectly fine using 'python manage.py runserver'. I can serve locally or passthrough to 0.0.0.0:8080 to review on mobile devices on my internal network.

I am keen to learn Tailwind css so took this opportunity to start implementing it into this project.

I have followed the documented installed and setup configuration. and with a build script and dev script I can get the static/css file to build correctly.

So great thus far. running  'python manage.py runserver' after each 'npm run build' implements css class edits and my pages reflect the new css styles as expected.

As I believe Tailwind css can serve pages while watching edits take place I am keen to implement this workflow.

In my package.jason file I have the standard dev script syntax "dev": 'npx tailwindcss -i ./src/input.css -o ./dist/output.css --watch',

With the paths to my input and output css files updated to real locations within my project.

However when I run watch nothing happens. 

npm run dev results in a terminal output of:

>tailwind@1.0.0 dev 
> npx tailwindcss -i ./src/input.css -o ./dist/output.css --watch

Rebuilding...
Done in 158ms.

If I edit pages within my project

the terminal outputs further

Rebuilding
Done in x-ms.

messages. and the tailwind css file updates.

But no page is available in Chrome at either 127.0.0.1:8000 or 127.0.0.1:5000, or 127.0.0.1:3000

I have clearly got something wrong in my configuration of Tailwindcss that is preventing watch from serving up the pages.

Any advice that will get me moving forward is greatly appreciated.

Kind regards

Duncan

**Comment by alex-krasikau** [comment, marked as answer]
Hello @DuncanGardiner!

> As I believe Tailwind CSS can serve pages while watching edits take place, I am keen to implement this workflow.

This is not correct. Tailwind processes your code and generates an output.css file that you need to add to the HTML generated by the Python server. This means that you still need to run python manage.py runserver and work on 0.0.0.0:8080 generated by the Python server.

To implement live updates, you can use the VS Code Live Server plugin. Check it out here: https://www.freecodecamp.org/news/vscode-live-server-auto-refresh-browser/

**Reply by DuncanGardiner** [reply]
Hi @alex-krasikau thanks for the very clear reply.

I did look at your suggested VSCode extension. However I was not able to configure that for Django apps in the time I have right now. Perhaps some wider reading is required on my part to get this going in the future.

That said, once I understood the need for a third-party browser reload-er and that ' manage.py runserver' was still required, I was able to find an appropriate solution.

I end up going down the route of installing a plugin to the Django project itself https://github.com/tjwalch/django-livereload-server .

This app requires that you run it, manage.py runserver and npm run dev is 3 separate terminals concurrently, but it does work well and stably. So for now I'll stick with it.

Thanks for the help on this question.

Kind regards

Duncan
